drivers/soundwire/cadence_master.c

```diff
@@ -1178,6 +1186,112 @@ static const struct sdw_master_port_ops cdns_port_ops = {
 	.dpn_port_enable_ch = cdns_port_enable,
 };
 
+/**
+ * sdw_cdns_suspend: Cadence PM suspend routine
+ *
+ * @cdns: Cadence instance
+ */
+int sdw_cdns_suspend(struct sdw_cdns *cdns, bool block_wake)
+{
+	u32 status;
+	int ret;
+
+	/* Check suspend status */
+	status = cdns_readl(cdns, CDNS_MCP_STAT);
+	if (status & CDNS_MCP_STAT_CLK_STOP) {
+		dev_dbg(cdns->dev, "Clock is already stopped\n");
+		return 1;
+	}
+
+	/*
+	 * In some system it is required to be able to put master
+	 * into a state in which it ignores wake-up trials in clock
+	 * stop state
+	 */
+	if (block_wake)
+		cdns_updatel(cdns, CDNS_MCP_CONTROL,
+			     CDNS_MCP_CONTROL_BLOCK_WAKEUP,
+			     CDNS_MCP_CONTROL_BLOCK_WAKEUP);
+
+	/*
+	 * when no slaves are attached to the bus, cmd accept
+	 * mode should be set to 1 to enter clock stop mode
+	 *
+	 */
+	if (list_empty(&cdns->bus.slaves))
+		cdns_updatel(cdns, CDNS_MCP_CONTROL,
+			     CDNS_MCP_CONTROL_CMD_ACCEPT, 1);
+	else
+		cdns_updatel(cdns, CDNS_MCP_CONTROL,
+			     CDNS_MCP_CONTROL_CMD_ACCEPT, 0);
```

Comment:
I don't think the test is correct.
This CMD_ACCEPT should be used when there are no devices attached on the link. If this is not set, if there is a broadcast write then the command ignored will be treated as a failure.

This should a scan that checks that each Slave is ATTACHED or ALERT, then you can change this bit to force a command ignored to be treated as 'no objection'